riscv/mmu.cc

```diff
@@ -161,6 +161,34 @@ void mmu_t::load_slow_path(reg_t addr, reg_t len, uint8_t* bytes, uint32_t xlate
   }
 }
 
+void mmu_t::clean_inval(reg_t addr, bool clean, bool inval)
+{
+  reg_t paddr = addr & ~(blocksz - 1);
+  try {
+    paddr = translate(paddr, blocksz, LOAD, 0);
+  } catch (trap_load_access_fault& t) {
+    trap_store_access_fault(t.has_gva(), addr, 0, 0);
+  } catch (trap_load_page_fault& t) {
+    trap_store_page_fault(t.has_gva(), addr, 0, 0);
+  } catch (trap_load_guest_page_fault& t) {
+    trap_store_guest_page_fault(addr, t.get_tval2(), 0);
```

Comment:
Is there any way this should share code with the AMOs, which do the [same thing here](https://github.com/riscv-software-src/riscv-isa-sim/blob/debdad850ba3c97171b027fc590db1e83cd9383a/riscv/mmu.h#L195-L207)?

Reply:
I'll try do do this.

Reply:
I have add amo_or_zicbom_op MACRO to support this:
```
  #define amo_or_zicbom_op(BODY) \
    try { \
      BODY \
    } catch (trap_load_address_misaligned& t) { \
      /* AMO faults should be reported as store faults */ \
      throw trap_store_address_misaligned(t.has_gva(), t.get_tval(), t.get_tval2(), t.get_tinst()); \
    } catch (trap_load_page_fault& t) { \
      /* AMO/Zicbom faults should be reported as store faults */ \
      throw trap_store_page_fault(t.has_gva(), t.get_tval(), t.get_tval2(), t.get_tinst()); \
    } catch (trap_load_access_fault& t) { \
      /* AMO/Zicbom faults should be reported as store faults */ \
      throw trap_store_access_fault(t.has_gva(), t.get_tval(), t.get_tval2(), t.get_tinst()); \
    } catch (trap_load_guest_page_fault& t) { \
      /* AMO/Zicbom faults should be reported as store faults */ \
      throw trap_store_guest_page_fault(t.get_tval(), t.get_tval2(), t.get_tinst()); \
    }
```